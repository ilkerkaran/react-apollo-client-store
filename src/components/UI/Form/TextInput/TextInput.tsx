import React from 'react';

const TextInput = ({
  className,
  inputName,
  label,
  defaultValue = undefined,
  isChanged = false,
  onChange
}: {className:string,
  inputName:string,
  defaultValue:string | undefined
  label:string,
  isChanged:boolean,
  onChange:(e:any) => any}) => (
    <div className={className}>
      <input
        className={isChanged ? 'Input Changed' : 'Input'}
        type="text"
        placeholder={label}
        onChange={onChange}
        name={inputName}
        defaultValue={defaultValue}
      />
    </div>
);

export default TextInput;
