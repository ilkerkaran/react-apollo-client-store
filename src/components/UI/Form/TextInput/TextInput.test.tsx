import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from './TextInput';

describe('<TextInput /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TextInput
        className=""
        defaultValue=""
        inputName=""
        isChanged={false}
        label="label"
        onChange={() => {}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
