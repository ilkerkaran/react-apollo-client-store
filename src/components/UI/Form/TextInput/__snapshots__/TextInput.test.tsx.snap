// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> snapshot  renders correctly 1`] = `
<div
  className=""
>
  <input
    className="Input"
    defaultValue=""
    name=""
    onChange={[Function]}
    placeholder="label"
    type="text"
  />
</div>
`;
